data-dashboard/lib/components/ButtonWithIcon/ButtonWithIcon.tsx

```diff
@@ -0,0 +1,57 @@
+import styled from 'styled-components';
+import React from 'react';
+import { CarbonIconType } from '@carbon/icons-react';
+
+interface Props {
+  text: string;
+  icon: CarbonIconType;
+  style?: React.CSSProperties;
+}
+
+const ButtonComponent = styled.button`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: center;
+
+  gap: 10px;
+
+  padding-left: 16px;
+  padding-right: 10px;
+
+  background-color: transparent;
```

Comment:
I prefer a white background color, it could be set here or given in the style prop from the Board parent component.